Dynamic credential paths fail on standby nodes - missing ForwardPerformanceStandby flag

## Description

Multiple secrets plugins are missing the `ForwardPerformanceStandby: true` flag on credential generation paths. This causes failures when requests are routed to standby nodes in HA clusters with horizontal scalability enabled, resulting in `cannot write to readonly storage` errors during lease registration.

## Environment
- OpenBao in HA cluster
- Standby reads enabled (horizontal scalability)
- Load balancer routing read operations to standby nodes

## Error Symptoms

When credential generation requests hit standby nodes:

[ERROR] core: failed to register lease: request_path=...

failed to persist lease entry: cannot write to readonly storage
an additional error was encountered deleting any lease associated with the newly-generated secret: failed to delete lease entry: cannot write to readonly storage


## Root Cause

Credential generation operations are defined as `logical.ReadOperation` but require writes:
1. Generate credentials (backend read)
2. **Create lease** (write to storage) Fails on read-only standby
3. **Store lease metadata** (write to storage) Fails on read-only standby

The OpenBao SDK provides `ForwardPerformanceStandby: true` specifically for this scenario, but it's missing from several plugins.

## Affected Plugins

### 1.  AWS Secrets Engine - **CONFIRMED NEEDS FIX**

**Files:**
- `secrets/aws/path_user.go` - Pattern: `(creds|sts)/{name}`
- `secrets/aws/path_static_creds.go` - Pattern: `static-creds/{name}`

**Fix Required:**
```
diff
 Operations: map[logical.Operation]framework.OperationHandler{
     logical.ReadOperation: &framework.PathOperation{
-        Callback: b.pathCredsRead,
+        Callback:                  b.pathCredsRead,
+        ForwardPerformanceStandby: true,
     },
     logical.UpdateOperation: &framework.PathOperation{
-        Callback: b.pathCredsRead,
+        Callback:                  b.pathCredsRead,
+        ForwardPerformanceStandby: true,
     },
 },
```

### 2. Consul Secrets Engine - CONFIRMED NEEDS FIX
File: secrets/consul/path_token.go
Pattern: creds/{role}
Evidence: Line ~115 uses b.Secret(SecretTokenType).Response(...) - creates lease

Fix Required:
```
-Callbacks: map[logical.Operation]framework.OperationFunc{
-    logical.ReadOperation: b.pathTokenRead,
-},
+Operations: map[logical.Operation]framework.OperationHandler{
+    logical.ReadOperation: &framework.PathOperation{
+        Callback:                  b.pathTokenRead,
+        ForwardPerformanceStandby: true,
+    },
+},
```

###  3. Nomad Secrets Engine - CONFIRMED NEEDS FIX
File: secrets/nomad/path_creds_create.go
Pattern: creds/{name}
Evidence: Line ~72 uses b.Secret(SecretTokenType).Response(...) - creates lease

Fix Required:
```
-Callbacks: map[logical.Operation]framework.OperationFunc{
-    logical.ReadOperation: b.pathTokenRead,
-},
+Operations: map[logical.Operation]framework.OperationHandler{
+    logical.ReadOperation: &framework.PathOperation{
+        Callback:                  b.pathTokenRead,
+        ForwardPerformanceStandby: true,
+    },
+},
```

### 4. GCP Secrets Engine - INVESTIGATION NEEDED
Files to review:

secrets/gcp/path_role_set_secrets.go - Patterns: roleset/{name}/key, roleset/{name}/token,
secrets/gcp/path_static_account_secrets.go - Patterns: static-account/{name}/key, static-account/{name}/token
secrets/gcp/path_impersonated_account_secrets.go - Pattern: impersonated-account/{name}/token
Note: The service account key paths call b.createServiceAccountKeySecret() which likely creates leases. The impersonated account token path returns data directly without b.Secret(), so it may not need the fix.

Needs confirmation on which paths actually create leases.

Related Work
This same issue was identified and fixed in the main OpenBao repository for built-in engines:

Database secret engine - path_creds_create.go
RabbitMQ secret engine - path_role_create.go
OpenLDAP secret engine - path_dynamic_creds.go

### Azure Secrets Engine already has the fix.

```
Operations: map[logical.Operation]framework.OperationHandler{
    logical.ReadOperation: &framework.PathOperation{
        Callback:                    b.pathSPRead,
        ForwardPerformanceSecondary: true,
        ForwardPerformanceStandby:   true,  //  Already present!
    },
},
```

Workaround
Until fixed, configure your load balancer to route credential generation paths to the active node only:

/v1/aws/creds/* → active node
/v1/aws/sts/* → active node
/v1/consul/creds/* → active node
/v1/nomad/creds/* → active node
/v1/gcp/roleset/*/key → active node (potentially)
/v1/gcp/roleset/*/token → active node (potentially)

References
SDK documentation: sdk/framework/path.go:186-209 in OpenBao repository
Framework handler: sdk/framework/backend.go:253 in OpenBao repository
Related OpenBao core issue: https://github.com/openbao/openbao/issues/2815

--- a/secrets/aws/secret_access_keys.go
+++ b/secrets/aws/secret_access_keys.go
@@ -11,6 +11,7 @@ import (
 
 	"github.com/hashicorp/go-secure-stdlib/awsutil/v2"
 	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/consts"
 	"github.com/openbao/openbao/sdk/v2/helper/template"
 	"github.com/openbao/openbao/sdk/v2/logical"
 
@@ -114,6 +115,12 @@ func (b *backend) getFederationToken(ctx context.Context, s logical.Storage,
 		policyARNs = append(policyARNs, groupPolicyARNs...)
 	}
 
+	// basic request validation is now done, but before we actually connect
+	// to AWS lets check, if we can even persist the lease in the end
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
+		return nil, logical.ErrReadOnly
+	}
+
 	stsClient, err := b.clientSTS(ctx, s)
 	if err != nil {
 		return logical.ErrorResponse(err.Error()), nil
@@ -204,6 +211,12 @@ func (b *backend) assumeRole(ctx context.Context, s logical.Storage,
 	}
 	if len(groupPolicyARNs) > 0 {
 		policyARNs = append(policyARNs, groupPolicyARNs...)
+	}
+
+	// basic request validation is now done, but before we actually connect
+	// to AWS lets check, if we can even persist the lease in the end
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
+		return nil, logical.ErrReadOnly
 	}
 
 	stsClient, err := b.clientSTS(ctx, s)

--- a/secrets/consul/path_token.go
+++ b/secrets/consul/path_token.go
@@ -11,6 +11,7 @@ import (
 
 	"github.com/hashicorp/consul/api"
 	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/consts"
 	"github.com/openbao/openbao/sdk/v2/logical"
 )
 
@@ -45,6 +46,12 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 	}
 	if entry == nil {
 		return logical.ErrorResponse(fmt.Sprintf("role %q not found", role)), nil
+	}
+
+	// basic request validation is now done, but before we actually connect
+	// to Consul lets check, if we can even persist the lease in the end
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
+		return nil, logical.ErrReadOnly
 	}
 
 	var roleConfigData roleConfig

--- a/secrets/gcp/secrets_service_account_key.go
+++ b/secrets/gcp/secrets_service_account_key.go
@@ -11,6 +11,7 @@ import (
 	"github.com/hashicorp/errwrap"
 	"github.com/hashicorp/go-gcp-common/gcputil"
 	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/consts"
 	"github.com/openbao/openbao/sdk/v2/logical"
 	"google.golang.org/api/iam/v1"
 )
@@ -161,6 +162,12 @@ func (b *backend) createServiceAccountKeySecret(ctx context.Context, s logical.S
 	}
 	if cfg == nil {
 		cfg = &config{}
+	}
+
+	// basic request validation is now done, but before we actually connect
+	// to GCP lets check, if we can even persist the lease in the end
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
+		return nil, logical.ErrReadOnly
 	}
 
 	iamC, err := b.IAMAdminClient(s)

--- a/secrets/nomad/path_creds_create.go
+++ b/secrets/nomad/path_creds_create.go
@@ -11,6 +11,7 @@ import (
 
 	"github.com/hashicorp/nomad/api"
 	"github.com/openbao/openbao/sdk/v2/framework"
+	"github.com/openbao/openbao/sdk/v2/helper/consts"
 	"github.com/openbao/openbao/sdk/v2/logical"
 )
 
@@ -65,6 +66,12 @@ func (b *backend) pathTokenRead(ctx context.Context, req *logical.Request, d *fr
 	}
 	if leaseConfig == nil {
 		leaseConfig = &configLease{}
+	}
+
+	// basic request validation is now done, but before we actually connect
+	// to Nomad lets check, if we can even persist the lease in the end
+	if b.System().ReplicationState().HasState(consts.ReplicationPerformanceStandby) {
+		return nil, logical.ErrReadOnly
 	}
 
 	// Get the nomad client

